Feature/check for faststart enabled

michaelcdillon opened the pull request:
I needed a way to only check if fast start had been applied on the media file, so I copy / pasta'd the first portion of the `fastStartImpl` method and added a check for a new `moovAtomEncountered` boolean at the end. 

I also added some functional tests with an 8kb test video for both methods, and finally, I upgraded gradle to the latest version (4.8).

Commit:
Added isFastStartEnabled check and tests.

- isFastStartEnabled returns a boolean if the file is fast start enabled
without needing an output file.
- added functional tests, and a quick 1s mp4 video for the tests to work
on.

Commit:
Upgraded to gradle 4.8 and bintray plubing 1.8.1

pszemus commented:
Coping/Pasting code instread of refactoring (e.g. introduce method) is never a good option.

Commit:
Refactored locate logic into standalone method.

- the logic to find ftyp and moov atoms is now in its own method.
- added hash validation on the test output files with fast start
enabled, comparing to a known hash value from the output of the
qtfaststart binary.

michaelcdillon commented:
Yes I agree, and I've refactored the locate logic into a method.

I did use some Java 8 Streams logic, a lambda, and some other current Java stuff which isn't necessary, but it is nice and clean code. If you want to keep the backward compatibility on this lib then please let me know and I'll remove it and also mark the source and target compatibility in the build.gradle file.

Some additional thoughts (not included in this PR):
- This could really benefit from including a real logging library like slf4j or something similar.
- The exceptions malformedFile & unsupportedFile could be thrown earlier inside the new locate method, but that would change existing behavior, so if you are open to another PR it would probably be wise to deprecate the old method, create a new one and throw those exceptions earlier.
